airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/file/object_storage/ObjectStorageClient.kt

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.file.object_storage
+
+import io.airbyte.cdk.load.message.RemoteObject
+import kotlinx.coroutines.flow.Flow
+
+interface ObjectStorageClient<T : RemoteObject> {
+    suspend fun list(prefix: String): Flow<T>
+    suspend fun put(key: String, bytes: ByteArray)
```

Comment:
doublechecking: this won't be on the code path for runtime, it's purely for the sake of writing `check` more easily? (and if yes, that's probably worth putting in a javadoc comment)

(i.e in an actual sync, we're never going to hold the entire file contents in memory, and presumably we would want a second function `put(key: String, localFile: File)`)

Reply:
It might be useful for small metadata files. If we stick with the staging directory model then we can keep a json file there with destination state until the platform supports it.

But yeah, next PR will have streaming upload support, which is what we'd want for the actual write.